Kernel/Bus/VirtIO/Device.cpp

```diff
@@ -125,6 +125,20 @@ UNMAP_AFTER_INIT void Device::initialize()
     }
 
     if (m_use_mmio) {
+        // Allocate backing kernel memory regions for BARs
+        for (auto bar = 0u; bar < sizeof(m_mmio)/sizeof(*m_mmio); ++bar) {
```

Comment:
There's no reason to initialise all BARs, the only bars we can access via the `mapping_for_bar` method are ones advertised by the various configs, so we should only initialize the bars referenced by `cfg->bar` for each `cfg` in `m_configs`

Reply:
Also, helpful reminder about `sizeof(foo)/sizeof(*foo)`: use `array_size(foo)` instead :)